cmd/fs-v1.go

```diff
@@ -113,12 +113,48 @@ func fsReaddirMetaBuckets(fsPath string) ([]string, error) {
 	return f.Readdirnames(-1)
 }
 
+// List of all bucket metadata configs.
 var bucketMetadataConfigs = []string{
 	bucketNotificationConfig,
 	bucketListenerConfig,
 	bucketPolicyConfig,
 }
 
+func migrateBucketMetadataConfigs(metaBucket, bucket, tmpBucket string) error {
+	for _, bucketMetaFile := range bucketMetadataConfigs {
+		_, err := fsStatDir(pathJoin(metaBucket, tmpBucket, bucketMetaFile))
+		if err != nil && errorCause(err) != errVolumeAccessDenied && errorCause(err) != errVolumeNotFound {
+			return err
+		}
+
+		// There are no such files or directories found, proceed
+		// to next bucket metadata config.
+		if errorCause(err) == errVolumeNotFound {
+			continue
+		}
+
+		// Bucket metadata is a file.
+		if errorCause(err) == errVolumeAccessDenied {
+			if err = fsRenameFile(pathJoin(metaBucket, tmpBucket, bucketMetaFile),
+				pathJoin(metaBucket, bucket, bucketMetaFile)); err != nil {
+				if errorCause(err) != errFileNotFound {
+					return err
+				}
+			}
+			continue
+		} // else { bucket metadata is a regular file.
+		if err = fsRenameFile(pathJoin(metaBucket, tmpBucket, bucketMetaFile),
+			pathJoin(metaBucket, bucket, objectMetaPrefix, bucketMetaFile)); err != nil {
+			if errorCause(err) != errFileNotFound {
+				return err
+			}
+		}
```

Comment:
the above if conditions body should be reversed, now this is what happens after migration:
```
krishna@escape:~$ ls export/.minio.sys/buckets/test/objects/
policy.json
krishna@escape:~$ 
```